src/System.ServiceModel.Tests.Common/src/testproperties.targets

```diff
@@ -0,0 +1,215 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- 
+    CreateTestPropertiesFile:
+      This task generates a C# file to initialize a dictionary with current MSBuild properties.
+      FileName:    the full name of the file to generate
+      PrefixNames: a semicolon-separated list of prefixes.  Only properties whose name begin
+                   with one of these prefixes will be extracted.
+  -->
+  <UsingTask TaskName="CreateTestPropertiesFile" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v12.0.dll">
+    <ParameterGroup>
+      <FileName ParameterType="System.String" Required="true" />
+      <PrefixNames ParameterType="System.String" Required="true" />
+    </ParameterGroup>
+    <Task>
+      <Reference Include="System" />
+      <Reference Include="System.IO" />
+      <Using Namespace="System.Reflection" />
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[
+        
+        // Project properties are available via Microsoft.Build.Execution.ProjectInstance.Properties.
```

Comment:
Reviewers: the use of Reflection to obtain the properties appears to be the only known way to do it.  The algorithm here shows up on StackOverflow as a known way to accomplish this.